File: le123.py
#!/usr/bin/env python3
# -*- coding:utf-8 -*- 
# author：yuanlang 
# creat_time: 2021/1/4 下午5:44
# file: le123.py


import frida, sys


def on_message(message, data):
    if message['type'] == 'send':
        print("[*] {0}".format(message['payload']))
    else:
        print(message)


jscode = """
Java.perform(function () {
    console.log("ok")
    var OkHttpClient = Java.use("okhttp3.OkHttpClient");
    OkHttpClient.newCall.implementation = function (request) {
        var result = this.newCall(request);
        console.log(Java.use("android.util.Log").getStackTraceString(Java.use("java.lang.Exception").$new()));
        console.log(request.toString());
        return result;
    }
    
    var SearchActivity = Java.use("com.elinkway.infinitemovies.ui.activity.SearchActivity");
})
"""

process = frida.get_usb_device().attach('com.le123.ysdq')
script = process.create_script(jscode)
script.on('message', on_message)
script.load()
sys.stdin.read()

File: le123.js
//
//Interceptor.attach(Module.findExportByName(null, "fopen"), {
//    onEnter: function(args) {
////        console.log("fopen Interceptor attached onEnter...");
////        console.log("fopen param0>>>>>>>" +  args[0].readCString());
//    },
//    onLeave: function(args) {
////        console.log("fopen Interceptor attached onLeave...");
//    }
//})
//
//var is_find_frida = false
//Interceptor.attach(Module.findExportByName(null, "strstr"), {
//    onEnter: function(args) {
////        console.log("strstr Interceptor attached onEnter...");
//        console.log("strstr param0>>>>>>>" +  args[0].readCString());
//        console.log("strstr param1>>>>>>>" +  args[1].readCString());
//        if (args[1].readCString() !=null && args[1].readCString().indexOf("frida")){
//            is_find_frida = true
//        }else{
//            is_find_frida = false
//        }
//    },
//    onLeave: function(retval) {
////        console.log("strstr Interceptor attached onLeave..." + retval);
//        if (is_find_frida){
//            retval.replace(ptr("0x0"))
//        }
//    }
//})

Java.perform(function () {

var OkHttpClient = Java.use("okhttp3.OkHttpClient");
OkHttpClient.newCall.implementation = function (request) {
    var result = this.newCall(request);
    console.log(Java.use("android.util.Log").getStackTraceString(Java.use("java.lang.Exception").$new()));
    console.log(request.toString());
    return result;
}

})
